### DISCUSSION TITLE
Tracking user caret

### DISCUSSION THREAD

**Discussion by alexfrench** [ether/etherpad-lite#4955] [Q&A] [answered]
Hi everyone.

I'm starting to look at Etherpad for a new application and wondered if someone could recommend the best way to track a user's caret position in a plugin? I've not seen an obvious hook at first look through the documentation, however, I'm sure I've simply missed something.

Thanks.

**Comment by JohnMcLear** [comment]
Cursor or caret?  Caret is flashing bar, cursor is mouse pointer.

**Reply by alexfrench** [reply]
Thanks fr=or your reply John. 

Caret. I want to know where the user is in the document, specifically as they move down through the document passing over the text and making changes.

**Reply by alexfrench** [reply]
And I think this is the hook I was looking for? aceSelectionChanged or aceEditEvent

**Reply by JohnMcLear** [reply]
Sorry I thought I answered this.  Etherpad does not disclose this information for privacy reasons.  As per @webzwo0i answer you can use a plugin to push this.

**Comment by webzwo0i** [comment, marked as answer]
Both will work. https://github.com/redhog/ep_cursortrace/ uses aceEditEvent to track other people's carets (it sends the new position to the server). aceEditEvent fires when the pad contents change, while aceSelectionChanged fires when the caret position changes.
